scipy/stats/_stats_py.py

```diff
@@ -7118,50 +7118,8 @@ def ttest_ind_from_stats(mean1, std1, nobs1, mean2, std2, nobs2,
     return Ttest_indResult(*res)
 
 
-def _ttest_nans(a, b, axis, namedtuple_type):
```

Comment:
This is no longer used, and the `_axis_nan_policy` approach makes it obsolete, so it can be removed. TBH, it looks like `_broadcast_shapes_with_dropped_axis` and `_shape_with_dropped_axis` can be removed, too.

Reply:
I agree. Feel free to remove those.

Reply:
I removed `_broadcast_shapes` and replaced it with `np.broadcast_shapes`, too. (`np.broadcast_shapes` was new in NumPy 1.20.0, so we didn't have that to work with.)